support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.jsx

```diff
@@ -0,0 +1,298 @@
+import { withStore } from 'components/subscriptionCheckouts/address/addressFields';
+import { countries } from 'helpers/internationalisation/country';
+
+// ----- Types ----- //
+
+type PropTypes = {|
+  ...FormFields,
+  country: IsoCountry,
+  signOut: typeof signOut,
+  submitForm: Function,
+  formErrors: FormError<FormField>[],
+  submissionError: ErrorReason | null,
+  productPrices: ProductPrices,
+  currencyId: IsoCurrency,
+  ...FormActionCreators,
+  csrf: Csrf,
+  payPalHasLoaded: boolean,
+  isTestUser: boolean,
+  amount: number,
+  billingPeriod: DigitalBillingPeriod,
+  setupRecurringPayPalPayment: Function,
+  validateForm: () => Function,
+  formIsValid: Function,
+  addressErrors: Array<Object>,
+  participations: Participations,
+|};
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state: CheckoutState) {
+  return {
+    ...getFormFields(state),
+    country: state.common.internationalisation.countryId,
+    formErrors: state.page.checkout.formErrors,
+    submissionError: state.page.checkout.submissionError,
+    productPrices: state.page.checkout.productPrices,
+    currencyId: state.common.internationalisation.currencyId,
+    csrf: state.page.csrf,
+    payPalHasLoaded: state.page.checkout.payPalHasLoaded,
+    paymentMethod: state.page.checkout.paymentMethod,
+    isTestUser: state.page.checkout.isTestUser,
+    amount: finalPrice(
+      state.page.checkout.productPrices,
+      state.common.internationalisation.countryId,
+      state.page.checkout.billingPeriod,
+    ).price,
+    billingPeriod: state.page.checkout.billingPeriod,
+    addressErrors: state.page.billingAddress.fields.formErrors,
+    participations: state.common.abParticipations,
+  };
+}
+
+// ----- Map Dispatch/Props ----- //
+function mapDispatchToProps() {
+  return {
+    ...formActionCreators,
+    formIsValid: () => (dispatch: Dispatch<Action>, getState: () => CheckoutState) => checkoutFormIsValid(getState()),
+    submitForm: () => (dispatch: Dispatch<Action>, getState: () => CheckoutState) =>
+      submitCheckoutForm(dispatch, getState()),
+    validateForm: () => (dispatch: Dispatch<Action>, getState: () => CheckoutState) => {
+      const state = getState();
+      validateCheckoutForm(dispatch, state);
+      // We need to track PayPal payment attempts here because PayPal behaves
+      // differently to other payment methods. All others are tracked in submit.js
+      const { paymentMethod } = state.page.checkout;
+      if (paymentMethod === PayPal) {
+        trackSubmitAttempt(PayPal, DigitalPack);
+      }
+    },
+    setupRecurringPayPalPayment: setupSubscriptionPayPalPayment,
+    signOut,
+  };
+}
+
+const TextAreaWithLabel = compose(asControlled, withLabel)(TextArea);
+const DatePickerWithError = withError(DatePickerFields);
+const Address = withStore(countries, 'billing', getBillingAddress);
+
+// ----- Component ----- //
+
+class DigitalCheckoutFormGift extends Component <PropTypes> {
```

Comment:
Do we absolutely have to use a class component here? All the other checkouts are function components

Reply:
No we don't. I have refactored it back to a function statement.